modules/dnn/test/test_layers.cpp

```diff
@@ -618,6 +618,44 @@ TEST(Layer_LSTM_Test_Accuracy_with_, HiddenParams)
     normAssert(h_t_reference, outputs[0]);
 }
 
+typedef testing::TestWithParam<tuple<int, int>> Layer_Softmax_1d_Test;
+TEST_P(Layer_Softmax_1d_Test, Accuracy) {
+
+    int batch_size = get<0>(GetParam());
+    int length = get<1>(GetParam());
+
+    LayerParams lp;
+    lp.type = "Softmax";
+    lp.name = "softmaxLayer";
+
+    Ptr<SoftmaxLayer> layer = SoftmaxLayer::create(lp);
+
+    // 1D input
+    std::vector<int> input_shape = {batch_size, length};
+    if (batch_size == 0)
+        input_shape.erase(input_shape.begin());
```

Comment:
Is (1, 1) considered as 1D?

Reply:
In my point of view both (L) and (N, L) can be considered as 1D tensors. There are examples when we apply layers on single dim Mat (e.g. constants folding) as well as 2-dimensional Mat (audio, tokens) are processed as 1D with batch.

Reply:
There is a mess which spreads over DNN module.

(L) could be the (N). Where element is a scalar (0D tensor) specified for N batches.

If we support bathes, when they should be specified everywhere (through dimensions (1, ) or separate flag which we don't have).

Reply:
> If we support bathes

I think we should drop this feature. It causes mess. Morden model format comes with proper batch description. It is what it is (that is to say we leave the problem to the model converter frontend so that we do not need to take care of it) and we do not offer the forced-batch feature anymore if we do want to support language models.